floodlight/models/geometry.py

```diff
@@ -209,3 +199,121 @@ def stretch_index(self, xy: XY, axis: str = None) -> TeamProperty:
         )
 
         return stretch_index
+
+
+class ConvexHullModel(BaseModel):
+    """Class for convex hull calculation.
+
+    Upon calling the :func:`~ConvexHullModel.fit`-method, this model creates a
+    ConvexHull object of the team for every frame.
+
+    Notes
+    -----
+    The convex hull is computed using the
+    `ConvexHull class
+    <https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.ConvexHull.
+    html>`_ from scipy.spatial and can be understood as the minimal convex area
+    containing all outfield players [reference].
+
+    The convex hull is also known in the literature under the terms ‘surface area’,
+    ‘coverage area’ and ‘playing area’ [1].
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from floodlight import XY
+    >>> from floodlight.models.geometry import ConvexHullModel
+
+    >>> xy = XY(np.array(   [0, 0, 10, 0, 10, 10, 0, 10, 5, 5], # square (10x10)
+    >>>                     [0, 0, 10, 0, 5, 10, 2, 2, 3, 3],   # triangle ((1/2)x10x10)
+    >>>                     [0, 5, 5, 10, 10, 5, 5, 0, 5, 5]))  # rhombus (5x10)
+    >>> exclude_xIDs = [4]
+    >>> chull = ConvexHullModel()
+    >>> chull.fit(xy=xy, exclude_xIDs=exclude_xIDs)
+    >>> chull.area_convex_hull()
+    TeamProperty(property=array([100, 50, 50]), name='area_convex_hull', framerate=None)
+
+    The area enclosed by the convex hull can be visualised using matplotlib as follows:
+
+    >>> import matplotlib.pyplot as plt
+    >>> c_hulls=[]
+    >>> c_hulls = chull.convex_hulls
+    >>> plt.plot(c_hulls[0].points[:, 0], c_hulls[0].points[:, 1], 'o')
+    >>> for simplex in c_hulls[0].simplices:
+    >>>     plt.plot(c_hulls[0].points[simplex, 0], c_hulls[0].points[simplex, 1], 'k-')
+    >>> plt.show()
+
+    .. image:: ../../_img/convex_hull.png
+
+    References
+    ----------
+        .. [1] `Low, B., Coutinho, D., Gonçalves, B., Rein, R., Memmert, D., & Sampaio,
+            J. (2020). A Systematic Review of Collective Tactical Behaviours in Football
+            Using Positional Data. Sports Medicine, 50(2), 343–385.
+            <https://link.springer.com/article/10.1007/s40279-019-01194-7>`_
+    """
+
+    def __init__(self):
+        super().__init__()
+        # model parameter
+        self.convex_hulls = None
```

Comment:
```suggestion
        self._convex_hulls_ = None
```
trailing underscore for fitted model parameters, otherwise the @requires_fit decorator won't work